tests/integration/test_sqs.py

```diff
@@ -508,3 +508,60 @@ def test_get_multiple_messages(self):
 
         # clean up
         self.client.delete_queue(QueueUrl=queue_url)
+
+    def test_lambda_invoked_by_sqs_message_with_delay_seconds(self):
+        function_name = 'lambda_func-{}'.format(short_uid())
+        queue_name = 'queue-{}'.format(short_uid())
+        delay_time = 10
+
+        queue_url = self.client.create_queue(QueueName=queue_name)['QueueUrl']
+        queue_arn = aws_stack.sqs_queue_arn(queue_name)
+
+        testutil.create_lambda_function(
+            handler_file=TEST_LAMBDA_ECHO_FILE,
+            func_name=function_name,
+            runtime=LAMBDA_RUNTIME_PYTHON36
+        )
+
+        lambda_log_group_name = '/aws/lambda/{}'.format(function_name)
+
+        lambda_client = aws_stack.connect_to_service('lambda')
+        lambda_client.create_event_source_mapping(
+            EventSourceArn=queue_arn,
+            FunctionName=function_name
+        )
+
+        rs = self.client.send_message(
+            QueueUrl=queue_url,
+            MessageBody='hello world.',
+            DelaySeconds=delay_time
+        )
+        message_id = rs['MessageId']
+
+        logs = aws_stack.connect_to_service('logs')
+
+        # There is no log group for this lambda (lambda not invoked yet)
+        rs = logs.describe_log_groups()
+        self.assertEqual(
+            len([lg['logGroupName'] for lg in rs['logGroups'] if lg['logGroupName'] == lambda_log_group_name]),
+            0
+        )
+
+        # After delay time, lambda invoked by sqs
```

Comment:
Is this delay actually caused by `DelaySeconds`, or simply because the Lambda triggering takes a couple of milliseconds in our infrastructure?

Perhaps we can add a `sleep(delay_time / 2)` in line 543, to make sure that the Lambda is actually waiting to be executed.

Also, I'd probably reduce `delay_time` in line 515 from `10` to `6`, or even `4`.